### ISSUE TITLE
Ternary net how to store the model

### ISSUE THREAD

**Issue by pinguo-luhaofang** [fengfu-chris/caffe-twns#1]

**Comment by fengfu-chris** [comment]
You need to store the ternary weights and the scaling factor alpha. For
X * (alpha * Wt) = (alpha * X) * Wt
this would hold the potential of speeding up the forward propagation.

**Comment by mathmanu** [comment]
@fengfu-chris 
Why not do it as:
(X * Wt) * alpha
So convolution is just add/substract and finally a scaling is done to the output.
Is my understanding correct - is it possible to implement it this way?

**Comment by fengfu-chris** [comment]
@mathmanu That is correct.  Both methods can be implemented with an additional Scale layer in Caffe.

**Comment by manwu1994** [comment]
> @fengfu-chris
> Why not do it as:
> (X * Wt) * alpha
> So convolution is just add/substract and finally a scaling is done to the output.
> Is my understanding correct - is it possible to implement it this way?

I also agree with ((X * Wt) * alpha) that speeding up the inference in hardware, do you have any method to extract the alpha parameters of each neurons?  Thank you...
